dvc/repo/reproduce.py

```diff
@@ -190,6 +194,10 @@ def _reproduce_stages(  # noqa: C901
             )
 
         try:
+            if kwargs.get("pull", False) and stage.is_data_source or stage.is_import:
+                logger.debug("Pulling '%s'", stage.addressing)
+                stage.repo.pull(stage.addressing)
```

Comment:
I'm -1 on using high-level API in `reproduce()`.

Reply:
@skshetry Could you elaborate more?

We currently call `stage.repo.cloud.pull` as part of `reproduce` (when `--pull` is passed), in the `restore` of the run cache which is way more down the rabbit hole:

https://github.com/iterative/dvc/blob/2a04263f021db190641ee2ca077b70563c74cdea/dvc/stage/cache.py#L201-L203

Is it better to "hide" it?

Reply:
`DataCloud.pull()` is an internal API and only concerns with pulling objects that you want, so it's a good fit to use here.

The only problem with it is that it does not support cloud versioning.

Reply:
> only concerns with pulling objects that you want

In the current P.R. I am using `repo.pull(dep.addressing)` so would only pull what I want.
Should I change to `cloud.pull` API? 
Still don't understand the benefit/reason, though

Reply:
`self.repo.cloud.pull()` has much smaller scope than `repo.pull()`, so it'll be faster and more reliable, and goes through less code. Also has less chances of getting broken in the future.